src/pages/fiscal-years/forms/FiscalYearForm.js

```diff
@@ -0,0 +1,187 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { validateNumberField } from 'src/lib/numberField-helper'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import CustomDatePicker from 'src/components/Inputs/CustomDatePicker'
+import { DataSets } from 'src/resources/DataSets'
+import { formatDateFromApi } from 'src/lib/date-helper'
+import { ControlContext } from 'src/providers/ControlContext'
+
+export default function FiscalYearForm({ labels, maxAccess, setStore, store }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const { recordId } = store
+
+  const invalidate = useInvalidate({
+    endpointId: SystemRepository.FiscalYears.page
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: null,
+      fiscalYear: '',
+      startDate: new Date(),
+      endDate: new Date(),
+      periods: '',
+      status: ''
+    },
+    maxAccess,
+    enableReinitialize: false,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      fiscalYear: yup
+        .number()
+        .transform((value, originalValue) => validateNumberField(value, originalValue))
+        .min(1900, labels.yearMin)
+        .max(2099, labels.yearMax),
+      startDate: yup.string().required(' '),
+      endDate: yup.string().required(' '),
+      periods: yup.string().required(' '),
+      status: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      const fiscalYear = obj.fiscalYear
+      await postRequest({
+        extension: SystemRepository.FiscalYears.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!recordId) {
+        toast.success(platformLabels.Added)
+        formik.setValues({
+          ...obj,
+          recordId: fiscalYear
```

Comment:
set fiscalYear returned from api response not from object

Reply:
it is not returned on post